docker_plugin/utils.py

```diff
@@ -50,24 +28,11 @@ def get_image_id(tag, image_id, client):
         raise NonRecoverableError(
             'Unable to get last created image: {0}'.format(e))
 
-    for img in images:
-        img_id, tags = (str(img.get('Id')), img.get('RepoTags'))
-        repotags = [repotag if tag in repotag else None for repotag in tags]
-        if filter(None, repotags) and img_id in str(image_id):
-            ctx.logger.debug(
-                'The tags and ids match, '
-                'assigning this image id: {0}.'.format(img_id))
-            return img_id
-        elif img_id in str(image_id):
-            ctx.logger.debug(
-                'No tags match, assigning this image id: {0}.'.format(img_id))
-            return img_id
-        elif possible_image_id in repotags:
-            ctx.logger.debug('stream_dict assigned the tag as image_id')
-            return img_id
-
-    raise NonRecoverableError(
-        'Unable to verify that the image id received during pull is valid.')
+    for image in images:
+        if '{0}:{1}'.format(repository, tag) in image.get('RepoTags'):
+            return image.get('Id')
+
+    raise NonRecoverableError('Unable to get image id.')
```

Comment:
the previous implementation seems to handle more configurations, is that on purpose?
also, the error message was more informative, I think.